modules/internal/ChapelHashtable.chpl

```diff
@@ -589,23 +601,32 @@ module ChapelHashtable {
             _moveInit(dstSlot.val, _moveToReturn(oldEntry.val));
 
             // move array elements to the new location
-            rehashHelpers.moveElementDuringRehash(oldslot, newslot);
+            if rehashHelpers != nil then
+              rehashHelpers!.moveElementDuringRehash(oldslot, newslot);
           }
         }
 
-        rehashHelpers.finishRehash(oldSize);
+        if rehashHelpers != nil then
+          rehashHelpers!.finishRehash(oldSize);
 
         // delete the old allocation
         _ddata_free(oldTable, oldSize);
 
       } else {
         // There were no entries, so just make a new allocation
 
-        // delete the old allocation
-        _ddata_free(table, tableSize);
-
         tableSizeNum = newSizeNum;
         tableSize = newSize;
+
+        if rehashHelpers != nil {
+          rehashHelpers!.startRehash(tableSize);
+          rehashHelpers!.finishRehash(oldSize);
+        }
+
+        // delete the old allocation
+        if table != nil then
+          _ddata_free(table, tableSize);
```

Comment:
Shouldn't that be `oldSize`?

Reply:
Yup, saw that too, and an upcoming commit will fix that. I'm going to try to get https://github.com/chapel-lang/chapel/issues/13661 in to make that easier to find too